src/templates/common/breadcrumbs/index.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  transformBreadcrumbs,
+  deriveLastBreadcrumbFromPath,
+  deriveLcBreadcrumbs,
+  filterInvalidCrumbs,
+} from '@/lib/utils/breadcrumbs'
+import {
+  BreadcrumbItem,
+  BreadCrumbLink,
+} from '@/types/dataTypes/drupal/field_type'
+
+interface BreadcrumbProps {
+  breadcrumbs?: BreadcrumbItem[]
+  breadcrumbTitle?: string
+  disableAnalytics?: boolean
+  deriveBreadcrumbsFromUrl?: boolean
+  replaceLastItem?: boolean
+  constructLcBreadcrumbs?: boolean
+  lcBreadcrumbsTitleInclude?: boolean
+  hideHomeBreadcrumb?: boolean
+  customCrumbHomeText?: string
+  entityPath?: string
+}
+const Breadcrumbs = ({
+  breadcrumbs,
+  entityPath,
+  breadcrumbTitle,
+  disableAnalytics,
+  deriveBreadcrumbsFromUrl,
+  replaceLastItem,
+  constructLcBreadcrumbs,
+  lcBreadcrumbsTitleInclude,
+  hideHomeBreadcrumb,
+  customCrumbHomeText,
+}: BreadcrumbProps) => {
+  if (!breadcrumbs) return null
+
+  if (customCrumbHomeText) {
+    breadcrumbs = breadcrumbs.map((crumb) => {
+      if (crumb.title === 'Home') {
+        return { ...crumb, title: customCrumbHomeText }
+      }
+      return crumb
+    })
+  } else if (hideHomeBreadcrumb) {
+    breadcrumbs = breadcrumbs.filter((crumb) => crumb.title !== 'Home')
+  }
+
+  if (deriveBreadcrumbsFromUrl) {
+    breadcrumbs = deriveLastBreadcrumbFromPath(
+      breadcrumbs,
+      breadcrumbTitle,
+      entityPath,
+      replaceLastItem
+    )
+  }
+
+  if (constructLcBreadcrumbs) {
+    breadcrumbs = deriveLcBreadcrumbs(
+      breadcrumbs,
+      breadcrumbTitle,
+      entityPath,
+      lcBreadcrumbsTitleInclude
+    )
+  }
+
+  const breadcrumbList: BreadcrumbItem[] = transformBreadcrumbs(breadcrumbs)
```

Comment:
I may have approved too early. I think this `BreadcrumbItems[]` might be in error. `transformBreadcrumbs` looks to return `BreadcrumbLink[]`, though that function doesn't have type definitions.

Reply:
My mistake. <3 typescript